src/Asv.Mavlink/Microservices/Ftp/Server/Ex/FtpServerEx.cs

```diff
@@ -477,9 +470,10 @@ public async Task WriteFile(WriteRequest request, Memory<byte> buffer, Cancellat
         var bytes = ArrayPool<byte>.Shared.Rent(request.Take);
```

Comment:
We don't need to create a buffer here. We can write directly to the input Memory<byte> e.g.: stream.ReadAsync(buffer)